### PR TITLE
Fix System32 module path duplication

### LINKED ISSUE #7327: System32 module path is added repeatedly in subprocesses
Steps to reproduce
------------------

```powershell
> $env:PSModulePath -split ';'
C:\Users\roholt\Documents\PowerShell\Modules
C:\Program Files\PowerShell\Modules
c:\program files\powershell\6-preview\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules

> & "$PSHOME\pwsh.exe"
> $env:PSModulePath -split ';'
C:\Users\roholt\Documents\PowerShell\Modules
C:\Program Files\PowerShell\Modules
c:\program files\powershell\6-preview\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules

> & "PSHOME\pwsh.exe"
> $env:PSModulePath -split ';'
C:\Users\roholt\Documents\PowerShell\Modules
C:\Program Files\PowerShell\Modules
c:\program files\powershell\6-preview\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules

> exit
> $env:PSModulePath -split ';'
C:\Users\roholt\Documents\PowerShell\Modules
C:\Program Files\PowerShell\Modules
c:\program files\powershell\6-preview\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules

> exit
> $env:PSModulePath -split ';'
C:\Users\roholt\Documents\PowerShell\Modules
C:\Program Files\PowerShell\Modules
c:\program files\powershell\6-preview\Modules
C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
```

Expected behavior
------------------

System32 module paths don't build up.

Actual behavior
---------------

System32 module paths keep getting tacked onto the end of the the PSModulePath.

Pros:
  * You always know how many PowerShells deep you are

Cons:
  * This is broken

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.1.0-preview.4
PSEdition                      Core
GitCommitId                    6.1.0-preview.4
OS                             Microsoft Windows 10.0.18200
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0


```

### PR DESCRIPTION
**Pull request by rjmholt**
## PR Summary

Resolves #7327. PowerShell Core subprocesses will only append the System32 module path to the PSModulePath when it is not already present.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1 [hunk 1]
```diff
@@ -324,30 +324,42 @@ Describe "Import-Module from CompatiblePSEditions-checked paths" -Tag "CI" {
...
 
-    It "Allows PowerShell Core 6 subprocesses to call core modules" -Skip:(-not $IsWindows) {
-        $errors = pwsh.exe -Command "Get-ChildItem" 2>&1 | Where-Object { $_ -is [System.Management.Automation.ErrorRecord] }
-        $errors | Should -Be $null
+    It "Does not duplicate the System32 module path in subprocesses" {
+        $sys32ModPathCount = pwsh.exe -C {
+            pwsh.exe -C '$null = $env:PSModulePath -match ([regex]::Escape((Join-Path $env:windir "System32" "WindowsPowerShell" "v1.0" "Modules"))); $matches.Count'
+        }
+
+        $sys32ModPathCount -le 1 | Should -BeTrue
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1]: Should it always be 1?
We could use `Should -BeLessOrEqual`.

**Review by iSazonov — APPROVED** [review]
With one minor comment

File: test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1 (hunk 1 above)
Comment (rjmholt) [reply test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1]: Good thought!

**Comment by anmenaga** [comment]
Restarted travis-ci test run.

**Review by anmenaga — APPROVED** [review]
LGTM

**Review by daxian-dbw — APPROVED** [review]

File: test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1 [hunk 2]
```diff
@@ -324,30 +324,42 @@ Describe "Import-Module from CompatiblePSEditions-checked paths" -Tag "CI" {
...
-            return
+        AfterAll {
+            if (-not $IsWindows)
+            {
+                return
+            }
+            Restore-ModulePath
+        }
+
+        It "Allows Windows PowerShell subprocesses to call `$PSHome modules still" -Skip:(-not $IsWindows) {
```
Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1]: `-Skip:(-not $IsWindows)` shouldn't be needed here as you have `$PSDefaultParameterValues` set at the beginning.

File: test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1 [hunk 3]
```diff
@@ -324,30 +324,42 @@ Describe "Import-Module from CompatiblePSEditions-checked paths" -Tag "CI" {
...
+            $errors = powershell.exe -Command "Get-ChildItem" 2>&1 | Where-Object { $_ -is [System.Management.Automation.ErrorRecord] }
+            $errors | Should -Be $null
         }
-        Restore-ModulePath
-    }
 
-    It "Allows Windows PowerShell subprocesses to call `$PSHome modules still" -Skip:(-not $IsWindows) {
-        $errors = powershell.exe -Command "Get-ChildItem" 2>&1 | Where-Object { $_ -is [System.Management.Automation.ErrorRecord] }
-        $errors | Should -Be $null
+        It "Allows PowerShell Core 6 subprocesses to call core modules" -Skip:(-not $IsWindows) {
```
Comment (daxian-dbw) [inline test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1]: same here.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**